File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        mp={0:-1}

        sum=0
        for i in range(len(nums)):
            sum+=nums[i];
            p=sum%k
            if p in mp:
                if i-mp[p]>1:
                    return True

            else:
                mp[p]=i


        return False        

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {

        unordered_map<int,int>ans;
        ans[0]=-1;
int sum=0;
        for(int i=0;i<nums.size();i++){
            sum+=nums[i];
            int p=sum%k;

            if(ans.find(p)!=ans.end() && i-ans[p]>=2 ){
                return true;
            }
            if(ans.find(p)==ans.end()){
                ans[p]=i;
            }

        }
        return false;
    }
};